tests/test_submain.py

```diff
@@ -1,47 +1,77 @@
+import mock
 import time
 
-from nose.tools import eq_, raises
+from nose.tools import eq_
 
-import validator.submain as submain
-from validator.errorbundler import ErrorBundle
+from validator import submain
 from validator.chromemanifest import ChromeManifest
-from validator.constants import *
-from helper import MockXPI
-
+from validator.errorbundler import ErrorBundle
+from .helper import MockXPI
 
-def test_prepare_package():
-    "Tests that the prepare_package function passes for valid data"
 
-    tp = submain.test_package
-    submain.test_package = lambda w, x, y, z, for_appversions: True
+@mock.patch("validator.submain.test_package")
+def test_prepare_package(test_package):
+    "Tests that the prepare_package function passes for valid data."
```

Comment:
Yeah, I didn't understand it either :( But rewriting all of the docstrings in this file wasn't an adventure I wanted to start down.